CMakeLists.txt

```diff
@@ -1386,6 +1390,15 @@ if(HPX_WITH_COMPILER_WARNINGS)
   endif()
 endif()
 
+# Configure compiler warnings as errors
+if(HPX_WITH_COMPILER_WARNINGS_AS_ERRORS)
+  if(MSVC)
+    hpx_add_compile_flag(-WX LANGUAGES C CXX Fortran)
+  else()
+    hpx_add_compile_flag_if_available(-Werror LANGUAGES CXX C Fortran)
+  endif()
+endif()
+
```

Comment:
This may break builds on MSVC as `-WX` will trigger tons of errors on superficial warnings generated in system headers (Boost). IIRC, `-WX` implies `-W4` which is the actual problem. But I will check again.

Reply:
Okay, thanks for checking. I'm happy to not enable this on Windows as well if it's not useful there. It is off by default though.